watcher/watcher.go

```diff
@@ -2,106 +2,157 @@ package watcher
 
 import (
 	"context"
-
-	"github.com/fsnotify/fsnotify"
-	"github.com/nhost/cli/logger"
+	"fmt"
+	"github.com/nhost/cli/nhost"
 	"github.com/nhost/cli/util"
 	"github.com/sirupsen/logrus"
+	"path/filepath"
+	"time"
 )
 
-type Operation func() error
-
-type Watcher struct {
-	log *logrus.Logger
-
-	//	It's inherently an fsnotify Watcher under the hood.
-	*fsnotify.Watcher
-
-	//  In the following format:
-	//  Key - Absolute File Name to Watch
-	//  Value - Function to execute
-	Map map[string]Operation
+type GitWatcher struct {
+	status        *util.Status
+	log           logrus.FieldLogger
+	repoExists    bool
+	branch        string
+	headRef       string
+	remoteRef     string
+	repoCreatedCh chan bool
```

Comment:
I think you can get away without channels leading to cleaner code. What I think you could do is the following:

1. Introduce methods like `onRepoCreation`, `onBranch` and `onRef` (better naming for the two latter might be great).
2. Instead of sending info to the cahnell you can just call those metods:

```
			if gw.branch != branch {
 				gw.branch = branch
 				gw.branchCh <- branch
 			}
```

vs

```
 			if gw.branch != branch {
         gw.OnBranch(branch)
 			}
```

3. Now you watch should be something like:

```
func (gw *GitWatcher) Watch(ctx context.Context, interval time.Duration, reloadFunc func(branch, ref string) error) {
 	go gw.watchRepoExists(ctx, interval)
 	go gw.watchBranchChange(ctx, interval)
 	go gw.watchRefChange(ctx, interval)

  <-ctx.Done()
  return
```